### PR TITLE
BookPageView UI and started bloc

### PR DESCRIPTION
**Pull request by UrosLITS**
Added necessary elements and created simple UI for now as well as 2 events for bloc for navigating through pages.Add,edit,delete yet to be created....

### CODE REVIEW

File: lib/presentation/common/custom_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  }) : super(key: key);
+
+  final String content;
+  final VoidCallback positiveAnswer;
+  final VoidCallback negativeAnswer;
+
+  @override
+  Widget build(BuildContext context) {
+    return AlertDialog(
+      content: SingleChildScrollView(
```
Comment (petarLits) [inline lib/presentation/common/custom_dialog.dart:19]: do we need SingleChildScrollView here?

**Review by petarLits — CHANGES_REQUESTED** [review]

File: lib/data/firebase_firestore/firebase_db_manager.dart [hunk 2]
```diff
@@ -62,4 +63,45 @@ class FirebaseDbManager {
...
+        if (item != null) {
+          bookPagesList.add(BookPages.fromJson(item));
+        }
+      }
+    }
+
+    final chapterRefs = db.collection(chaptersCollection).doc(bookID);
+    final snapShoots =
+        await chapterRefs.get().timeout(Duration(seconds: 3), onTimeout: () {
+      throw Exception("Cant reach server");
```
Comment (milicaLits) [inline lib/data/firebase_firestore/firebase_db_manager.dart:90]: Replace with constant timeoutErrorMessage

File: lib/presentation/bookPages/bloc/book_page_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+class BookPagesBloc extends Bloc<BookPagesEvents, BookPagesState> {
+  BookPagesBloc() : super((InitialState())) {
+    on<NextPageEvent>(_onNextPage);
+    on<PreviousPageEvent>(_onPreviousPage);
+  }
+
+  Future<void> _onNextPage(
+      NextPageEvent event, Emitter<BookPagesState> emit) async {
+    emit(NextPage(currentIndex: event.currentIndex + 1));
```
Comment (milicaLits) [inline lib/presentation/bookPages/bloc/book_page_bloc.dart:13]: You don't have to forward current index through events and state for next and previous page, you can do the calculation in listener of BlocConsumer.

**Review by milicaLits — CHANGES_REQUESTED** [review]

File: lib/presentation/bookPages/widgets/build_page_body.dart [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  Widget build(BuildContext context) {
+    return Container(
+      child: FloatColumn(
+        children: [
+          SizedBox(
+            height: 10,
+          ),
+          Floatable(
+            float: FCFloat.start,
+            child: Text(""),
```
Comment (lecicdjuro) [inline lib/presentation/bookPages/widgets/build_page_body.dart:28]: why empty text?

File: lib/presentation/common/custom_dialog.dart [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            Text(
+              content,
+            ),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+              children: [
+                TextButton(
+                  onPressed: positiveAnswer,
```
Comment (lecicdjuro) [inline lib/presentation/common/custom_dialog.dart:30]: just do Navgaitor.pop(true)

File: lib/presentation/common/custom_dialog.dart [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                  child: Text(AppLocalizations.of(context)!.yes),
+                ),
+                TextButton(
+                  onPressed: negativeAnswer,
```
Comment (lecicdjuro) [inline lib/presentation/common/custom_dialog.dart:34]: just do Navgaitor.pop(false)

Comment (lecicdjuro) [reply lib/presentation/common/custom_dialog.dart:34]: I understand that you use on it when you are developing Android, but this is bad practice for Flutter widgets

**Review by lecicdjuro — CHANGES_REQUESTED** [review]

File: lib/presentation/book/view/home_page_view.dart [hunk 7]
```diff
@@ -66,20 +66,15 @@ class _HomeBookPageView extends State<HomePageView> {
...
-                          });
-                    });
+                  context: context,
+                  builder: (BuildContext context) {
+                    return CustomDialog(
+                        content: AppLocalizations.of(context)!.log_out);
+                  },
+                );
                 if (result == true) {
+                  FirebaseAuthSingleton.instance.auth.signOut();
```
Comment (lecicdjuro) [inline lib/presentation/book/view/home_page_view.dart:76]: what if this async call fails?

Comment (lecicdjuro) [inline lib/presentation/book/view/home_page_view.dart:76]: make event

**Review by lecicdjuro — CHANGES_REQUESTED** [review]

**Review by lecicdjuro — APPROVED** [review]

**Pull request merged**